aten/src/ATen/native/cuda/ReplicationPadding.cu

```diff
@@ -75,6 +71,135 @@ __global__ void replication_pad_backward_kernel(
   atomicAdd(&gradInput[batch][plane][inputPointX], valueToCopy);
 }
 
+template <typename scalar_t>
+__global__ void replication_pad_forward_kernel(
```

Comment:
nit: Rather than have multiple overloads for replication pad, it would be better if they explicitly identified the dimensionality of the operation in question. Makes it easier to search around the code.